### PR TITLE
[JEP-222] Pass JENKINS_WEB_SOCKET when so configured

### PR DESCRIPTION
**Pull request by jglick**
See [JEP-222](https://jenkins.io/jep/222). Downstream of https://github.com/jenkinsci/jenkins/pull/4369 and presumes https://github.com/jenkinsci/docker-jnlp-slave/pull/130.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -47,7 +47,7 @@
     <jenkins.host.address />
     <slaveAgentPort />
     <java.level>8</java.level>
-    <jenkins.version>2.190.1</jenkins.version>
+    <jenkins.version>2.217</jenkins.version>
```
Comment (jglick) [inline pom.xml]: Could be made compatible with 2.204.x if desired by using a bit of reflection.

**Comment by jglick** [comment]
Test failure is `Failed to bind to /0.0.0.0:8000` because `Address already in use`. I suppose Kind is exposing the service ports to the outside, and this fails with multiple builds run in one VM or something? I do not see an easy way to randomize it or bind to an open socket.

**Comment by jglick** [comment]
Though [documentation](https://kind.sigs.k8s.io/docs/user/configuration/#extra-port-mappings) implies that nothing gets exposed by default, so I would have expected the 8000 port (also 50000) to be private to this Kind network.

**Comment by jglick** [comment]
> randomize it

#709

**Comment by Vlatombe** [comment]
Looks like CI is not happy.

`runInPod` looks like a flake, but `runWithEnvVariables` looks legit

**Comment by jglick** [comment]
It ought to be possible to write a test to get some coverage for WebSocket mode. Will try it.

**Comment by jglick** [comment]
Kind flake?

```
Error from server (ServerTimeout): error when creating "STDIN": No API token found for service account "jenkins", retry after the token is automatically created and added to the service account
```

**Comment by Vlatombe** [comment]
> Kind flake?

Yes, seen that one before.

**Comment by jglick** [comment]
Took 30m just to download dependencies, trying again, sigh

**Comment by jglick** [comment]
Another apparent flake:

```
WARNING	o.c.j.p.k.KubernetesClientProvider#closeExpiredClients: High number of expired clients (56), may cause memory leaks
```

Should maybe be excluded from `#runInPod`’s assertion, or maybe this should just be downgraded to `INFO`.

File: pom.xml (hunk 1 above)
Comment (micxer) [reply pom.xml]: Is this necessary to have Websockets enabled? This feature would help us in our setup a lot but we are on LTS so we cannot upgrade the plugin to version 1.25. 😢

Comment (Vlatombe) [reply pom.xml]: Wait for the next LTS.

Comment (jglick) [reply pom.xml]: 2.222.1 should be available within a month or so.

**Pull request merged**